### DISCUSSION TITLE
Opening an edit-modal programmatically

### DISCUSSION THREAD

**Discussion by MaartenJC** [filamentphp/filament#5532] [Help]
Hello,
how can i open an edit-modal programmatically outside the Table Builder?

For example: with wire-elements/modal it can be done like this:
`onclick='Livewire.emit("openModal", "path.to.edit",<?php echo json_encode([$record->id]) ?>); return false;'`

I found how to open a static modal like this:
`x-on:click="$dispatch('open-modal', { id: 'custom-modal'}); return false;"`

But how can i pass a record ID?

Thanks.

**Comment by divdax** [comment]
Have you found a solution?

**Comment by nathanjansen** [comment]
Maybe this helps: https://filamentphp.com/docs/3.x/support/blade-components/modal

<img width="858" alt="image" src="https://github.com/filamentphp/filament/assets/73473323/3f137280-e7d8-4d48-8e38-41c7fb942ac3">
